internlm/utils/common.py

```diff
@@ -51,21 +51,17 @@ def _move_tensor(element):
         for idx, item in enumerate(element):
             if isinstance(item, dict):
                 for key, value in item.items():
-                    assert not value.is_cuda, "elements are already on devices."
                     item[key] = value.to(get_current_device()).detach()
             elif isinstance(item, list):
                 for index, value in enumerate(item):
-                    assert not value.is_cuda, "elements are already on devices."
                     item[index] = value.to(get_current_device()).detach()
             elif torch.is_tensor(item):
-                if not item.is_cuda:
-                    element[idx] = item.to(get_current_device()).detach()
+                element[idx] = item.to(get_current_device()).detach()
             else:
                 assert False, f"{type(item)}, {item}"
     else:
         assert torch.is_tensor(element), f"element should be of type tensor, but got {type(element)}"
-        if not element.is_cuda:
```

Comment:
<img width="693" alt="image" src="https://github.com/InternLM/InternEvo/assets/6859848/a17393bd-44be-474e-adb7-09fa25b85d6e">
maybe we just need to judge whether the tensor is on CPU